03week/towersOfHanoi.js

```diff
@@ -7,36 +7,54 @@ const rl = readline.createInterface({
   output: process.stdout
 });
 
+//This is the initial setting at the beginning of towersOfHanoi.  Part of premade code
 let stacks = {
   a: [4, 3, 2, 1],
   b: [],
   c: []
 };
 
-function printStacks() {
+//This function prints the stacks on to the screen.  Part of premade code
+let printStacks() {
   console.log("a: " + stacks.a);
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
+//printBeginningStacks, reset the var stacks to its original position, nest stacks inside
+const printBeginningStacks=(stacks)=> {}
+  let stacks = {
+    a: [4, 3, 2, 1],
+    b: [],
+    c: []
+  };
+//movePiece(), Moves the last index of the startStack to the last index of the endStack, lastindexof method
+const movePiece=(startstack, endStack)=> {
+    endStack.push(startstack.lastindexof)
+    startstack.lastindexof.slice
+  }
 
+//isLegal(), checks to see if the last index of the start stack is less than the end stack last index.  If the value is less, run movePiece.  If the value is greater, deny the move, lastindexof method
+const isLegal=(startstack, endStack)=> {
+  if((startstack.lastindexof < endStack.lastindexof) || (startstack.lastindexof > 0)) {
+    movePiece
+  } else {
+    return 'Not a legal move.';
+  }
 }
 
-function isLegal() {
-  // Your code here
-
-}
-
-function checkForWin() {
-  // Your code here
-
+//checkForWin, if all of the numbers are in the last stack, run the win sequence, which should reset the object, if statement
+const checkForWin=()=>
+  if (stacks === [a: [], b: [1], c:[4, 3, 2, 1]]) {
```

Comment:
I like where your'e going here, however in Javascript, you can't directly compare two objects or arrays. The reason being is that every object is different. So if you made an object const a ={} and const b = {}, Javascript will never evaluate a === b. So knowing that, what is a different way that you could check for a win?